File: POH7/Mizugi/Main.rb
#! ruby
# Try POH
# author: Leonardone @ NEETSDKASU

def gs(); gets.chomp; end
def gi(); gets.to_i; end
def gss(); gs.split; end
def gis(); gss.map(&:to_i); end

n = gi

MD = 10 ** 9

r, b = (1..n).inject([1,0]) {|(r,b), v|
    (b += 1; v /= 2) while v%2==0
    (b -= 1; v /= 5) while v%5==0
    [r * v % MD, b]
}

b.times do
    r=r * 2 % MD
end

puts r

File: POH7/Mizugi/Main.cs
// Try POH
// author: Leonardone @ NEETSDKASU
using System;
using System.Linq;

public class Solver
{
    public static void Main()
    {
        long r = 1, n = gl();
        int c = 0;
        
        while (n > 1)
        {
            var x = n--;
            while (x % 5 == 0)
            {
                x /= 5;
                c--;
            }
            while (x % 2 == 0)
            {
                x /= 2;
                c++;
            }
            r = (r * x) % 1000000000;
        }
        
        while (c-- > 0)
        {
            r = (r * 2) % 1000000000;
        }
        
        Console.WriteLine(r);
    }
    
    ///////////////////////////////////////////////////////////////////
    static string   gs()  { return Console.ReadLine(); }
    static int      gi()  { return int.Parse(gs()); }
    static long     gl()  { return long.Parse(gs()); }
    static string[] gss() { return gs().Split(" ".ToCharArray()); }
    static int[] gis() { return gss().Select( int.Parse ).ToArray(); }
}
